src/metaprob/state.clj

```diff
@@ -35,15 +35,14 @@
     (cond (seq? state) (first state)
           (vector? state) (assert false "no value")
           (map? state)
-          (let [value (get state :value :no-value)]
-            (assert (not (= value :no-value)) ["state has no value" state])
-            value)
-          true (assert false ["not a state" state]))))
+          (do (assert (contains? state :value) ["state has no value" state])
+              (get state :value))
```

Comment:
Not keen on this because it does the `get` twice.  This is seriously inner loop code and while it might be hard to measure the difference in time, or even detect it, seeing the redundant work makes me uncomfortable.

Reply:
Makes sense. Backing this change out.